h5py/h5d.pyx

```diff
@@ -497,3 +503,92 @@ cdef class DatasetID(ObjectID):
                     H5Sclose(space_id)
 
             return filters, bytes(data)
+
+    IF HDF5_VERSION >= (1, 10, 5):
+
+        @with_phil
+        def get_num_chunks(self, SpaceID space=None):
+            """ (SpaceID space=None) => INT num_chunks
+
+            Retrieve the number of chunks that have nonempty intersection with a
+            specified dataspace. Currently, this function only gets the number
+            of all written chunks, regardless of the dataspace.
+
+            Feature requires: 1.10.5 HDF5
+            """
+            cdef hsize_t num_chunks
+
+            if space is None:
+                space = self.get_space()
+            H5Dget_num_chunks(self.id, space.id, &num_chunks)
+            return num_chunks
+
+        @with_phil
+        def get_chunk_info(self, hsize_t index, SpaceID space=None):
+            """ (hsize_t index, SpaceID space=None) => StoreInfo
+
+            Retrieve storage information about a chunk specified by its index.
+
+            Feature requires: 1.10.5 HDF5
+            """
+            cdef haddr_t byte_offset
+            cdef hsize_t size
+            cdef hsize_t *chunk_offset
+            cdef unsigned filter_mask
+            cdef hid_t space_id = 0
+            cdef int rank
+
+            if space is None:
+                space_id = H5Dget_space(self.id)
+            else:
+                space_id = space.id
+
+            rank = H5Sget_simple_extent_ndims(space_id)
+            chunk_offset = <hsize_t*>emalloc(sizeof(hsize_t) * rank)
+            H5Dget_chunk_info(self.id, space_id, index, chunk_offset,
+                              &filter_mask, &byte_offset, &size)
+            cdef tuple cot = convert_dims(chunk_offset, <hsize_t>rank)
+            efree(chunk_offset)
+
+            if space is None:
+                H5Sclose(space_id)
+
+            return StoreInfo(index,
```

Comment:
I think we should drop `index` from the return value. For `get_chunk_info` it's an input parameter, so the caller already knows it, and for `get_chunk_info_by_coord` it's not set.

Reply:
Keeping `index` in the same place could help simplify the code that uses chunk storage info.

Reply:
I'm not sure what you mean.

In case it wasn't clear, I'm suggesting that we remove `index` from the `StoreInfo` named tuple. It's only used as a return value from these two methods. One of them doesn't have a value for it, and for the other it's an input to the method. So it seems more consistent not to return it.

Reply:
Sorry for being terse in my reply.  Is there any damage keeping `index` in `StoreInfo`? You are correct that it is not vital but it will stay with all the other chunk storage information. Bear in mind, also, that the chunk storage API does not seem final since there is a placeholder for a dataspace which is currently unused.

Reply:
It seems like an unnecessary complication. If you get a `StoreInfo` tuple from an unknown source, you can't assume it has a useful index, because `get_chunk_info_by_coord` doesn't return one. So you effectively have two types of `StoreInfo`, one with an index, one without. And any code that gets the *with index* kind has passed the same index in, so it's not necessary.

If this was already in a released version, it would be a pain to change. But it's not, so we can iron out this kind of API wart before it's in a release.

Reply:
Okay, removed `index`.